distributed/worker_memory.py

```diff
@@ -401,7 +401,15 @@ def parse_memory_limit(
     assert isinstance(memory_limit, int)
     if memory_limit == 0:
         return None
-    return min(memory_limit, system.MEMORY_LIMIT)
+    if system.MEMORY_LIMIT < memory_limit:
+        logger.warning(
+            "Ignoring provided memory limit %s due to system memory limit of %s",
```

Comment:
Can we add a small test for this? It'd be good to make sure we don't loose this informative message in the future